### DISCUSSION TITLE
Decentralized ISP «Medium»

### DISCUSSION THREAD

**Discussion by ghost** [PurpleI2P/PurpleI2P#10] [Ideas]
Hi there,

I am an operator of multiple «Medium» access points in Russia. «Medium» uses I2P as primary transport.

You can get more info about «Medium» [here](https://habr.com/ru/post/459386/).

«Medium» has a own [DNS-like system](http://dns3a44zzbexpzpqakipnshxvdovx5vx4wpwknncbcekldjqp2wa.b32.i2p/) and [certification authority](https://github.com/medium-isp/medium-pki).

Maybe publishing some info about «Medium» in I2P related resources is a good idea? It will be wery helpful to grow the network.

**Comment by orignal** [comment]
Ну мы вообще то:
1. Говорим по русски
2. Читаем хабр

>Maybe publishing some info about «Medium» in I2P related resources is a good idea?  

Уже

**Comment by ghost** [comment]
Ух ты! Земляки!

> Уже

Уже - это где?

Кстати: я на досуге сделал сайт себе в стиле Веб 1.0. И даже не поленился создать плашки:

![](http://i.imgur.com/rTaVp4c.png)
![](http://i.imgur.com/jxMJ8rY.png)

На ваш суд!

**Comment by ghost** [comment]
Собственно, [сам сайт](http://veakv6mt2qs577mkgnozk7o6tunv5c7ptvfr6iwnxvt4nxilo4xa.b32.i2p/)

**Comment by orignal** [comment]
>Уже - это где?

Сэди в твитере про вас писала.

**Comment by ghost** [comment]
@original, не являюсь завсегдатаем Твиттера, так что не знаю, кто это. Просветите?

**Comment by l-n-s** [comment]
https://twitter.com/search?q=%23i2p&src=typed_query&f=live тут можно найти

**Comment by ghost** [comment]
@l-n-s, приветствую! Когда-то читал Вашу публикацию про XMPP в I2P на Хабре. Очень рад Вас здесь встретить.

**Comment by l-n-s** [comment]
Спасибо :)

А у вас в medium есть XMPP сервис?

**Comment by ghost** [comment]
Пока что нет, только IRC.

**Comment by r4sas** [comment]
> «Medium» has a own [DNS-like system](http://dns3a44zzbexpzpqakipnshxvdovx5vx4wpwknncbcekldjqp2wa.b32.i2p/) and [certification authority](https://github.com/medium-isp/medium-pki).

В обоих ссылках (как на ваш "DNS", так и ссылка на PKI) при обращении к b32 идет редирект на домены, с https, при этом ни где регистрации имён нет - ни на inr, ни stats, ни даже банальном identiguy.
В итоге - не юзабельно. Вообще.

**Comment by ghost** [comment]
@r4sas, всё правильно. Ни на inr, ни на stats мы не публикуемся. Все операторы Медиума тянут подписку с [dns.medium.i2p](http://dns3a44zzbexpzpqakipnshxvdovx5vx4wpwknncbcekldjqp2wa.b32.i2p/).

**Comment by ghost** [comment]
@r4sas, DNS не требует обязательного использования HTTPS. Возможно, у Вас HSTS от старых хостов Медиума?

![](http://i.imgur.com/URofRpM.png)

**Comment by r4sas** [comment]
К которой не получить доступа никаким образом...

http://dns3a44zzbexpzpqakipnshxvdovx5vx4wpwknncbcekldjqp2wa.b32.i2p/
=>
http://dns.medium.i2p/export/alive-hosts.txt
=>
![image](https://user-images.githubusercontent.com/7023612/61596568-f961cf80-ac0d-11e9-8968-1c5fc9da01a4.png)

**Comment by ghost** [comment]
@r4sas, у Вас нет доступа по .b32 ссылке?

**Comment by r4sas** [comment]
Переношу иссуй в community.

**Comment by r4sas** [comment]
@podivilov как видите выше - нет.

**Comment by ghost** [comment]
@r4sas, но выше у Вас в омнибоксе браузера указана ссылка на dns.medium.i2p, не .b32. Если нет доступа по .b32, то это очень странно.

**Comment by orignal** [comment]
И у меня нет доступа по сссылке той

**Comment by r4sas** [comment]
> @r4sas, но выше у Вас в омнибоксе браузера указана ссылка на dns.medium.i2p, не .b32. Если нет доступа по .b32, то это очень странно.

Я же указал цепочку вашего редиректа.

**Comment by ghost** [comment]
@r4sas, виноват, неправильно Вас понял.

![](http://i.imgur.com/WzgtpAT.png)

Проблема в редиректе, который не принимает во внимание хост. Сейчас исправлю.

**Comment by ghost** [comment]
@r4sas, исправил. Можете проверять.

![](http://i.imgur.com/Uw0AJsl.png)

**Comment by r4sas** [comment]
Теперь все работает.

**Comment by majestrate** [comment]
I have something you may find useful (my IP packet onion router) https://github.com/loki-project/loki-network

**Comment by ghost** [comment]
@majestrate, IDK how Loki can help Medium to grow up. Maybe you can provide some examples?

**Comment by majestrate** [comment]
@podivilov loki**net** is an ip layer onion router that was made to run on soho routers, you can push all lan traffic over it since it exposes a tun interface instead of providing application tunnels. you don't need the coin part to run a network.

**Comment by ghost** [comment]
@majestrate Sounds interesting. You can create issue [here](https://github.com/medium-isp/medium/issues).

**Comment by ghost** [comment]
Анонс Medium Summer Meetup 2019

https://habr.com/ru/post/461123/

@orignal, @l-n-s, @r4sas! Если вы в Москве или в области -- буду рад вас видеть на встрече!
